src/main/scala/workflow/graph/Estimator.scala

```diff
@@ -0,0 +1,60 @@
+package workflow.graph
+
+import org.apache.spark.rdd.RDD
+
+/**
+ * An estimator has a `fitRDD` method which takes an input and emits a [[Transformer]]
+ * @tparam A The type of input this estimator (and the resulting Transformer) takes
+ * @tparam B The output type of the Transformer this estimator produces when being fit
+ */
+abstract class Estimator[A, B] extends EstimatorOperator {
+  /**
+   * Constructs a pipeline that fits this estimator to training data,
+   * then applies the resultant transformer to the Pipeline input.
+   *
+   * @param data The training data
+   * @return A pipeline that fits this estimator and applies the result to inputs.
+   */
+  final def withData(data: RDD[A]): Pipeline[A, B] = {
+    withData(PipelineDatasetOut(data))
+  }
+
+  /**
+   * Constructs a pipeline that fits this estimator to training data,
+   * then applies the resultant transformer to the Pipeline input.
+   *
+   * @param data The training data
+   * @return A pipeline that fits this estimator and applies the result to inputs.
+   */
+  final def withData(data: PipelineDatasetOut[A]): Pipeline[A, B] = {
+    // Remove the data sink,
+    // Then insert this estimator into the graph with the data as the input
+    val curSink = data.executor.graph.getSinkDependency(data.sink)
+    val (estGraph, estId) = data.executor.graph.removeSink(data.sink).addNode(this, Seq(curSink))
+
+    // Now that the estimator is attached to the data, we need to build a pipeline DAG
+    // that applies the fit output of the estimator. We do this by creating a new Source in the DAG,
+    // Adding a delegating transformer that depends on the source and the label estimator,
```

Comment:
Nit: Can we move the comment next to the appropriate line of code in the next 4 lines ?